### ISSUE TITLE
Gzip support?

### ISSUE THREAD

**Issue by andylokandy** [elba/elba#32]
Is't worth supporting gzip? This can significantly reduce the network usage. We can only support it on http transferring stage, but we can also support it as a new resolution as `tar+gzip`.

**Comment by dcao** [comment]
Ideally we could re-use the tar resolution and detect if the retrieved file has been compressed with gzip; either way I think this is a feature worth implementing

**Comment by andylokandy** [comment]
Good point! So I'm gonna to implement gzip as a strong defaut in backend.
